wolfcrypt/src/asn.c

```diff
@@ -5612,8 +5642,8 @@ static int DecodePolicyOID(char *out, word32 outSz, byte *in, word32 inSz)
             return ASN_PARSE_E;
         }
 
-        /* Validate total length (2 is the CERT_POLICY_OID+SEQ) */
-        if ((total_length + 2) != sz) {
+        /* Validate total length */
+        if (total_length > (sz - (int)idx)) {
```

Comment:
This section of code was originally changed in PR #336, where I resolved an issue parsing multiple items. It was later changed through a few refactors and now the comment about `2 is the CERT_POLICY_OID+SEQ` is incorrect here. The first sequence is total length. Then its a Sequence and OID in the loop. The updates in this PR are correct for handling checking max size.